### ISSUE TITLE
DASH Downloader doesn't work with specific .mpd

### ISSUE THREAD

**Issue by pi-andrei** [google/ExoPlayer#3729]
### Issue description
When trying to download movie with Dash Downloader it fails with error:
"Unbounded index for representation: ..."

Seems, that issue caused by these part of code in DashDownloader.java class in getSegments method:
```
int segmentCount = index.getSegmentCount(C.TIME_UNSET);
if (segmentCount == DashSegmentIndex.INDEX_UNBOUNDED) {
    throw new DownloadException("Unbounded index for representation: " + key);
}
```

If `C.TIME_UNSET` is always used as parameter, then BaseSegment code rely on `<SegmentTimeline>` tag in mpd file to get segments count. But there is no `<SegmentTimeline>` tag in specific mpd.

As a workaround I replaced above code with next lines, and download started successfully:
```
long periodDurationUs = manifest.getPeriodDurationUs(key.periodIndex);
int segmentCount = index.getSegmentCount(periodDurationUs);
if (segmentCount == DashSegmentIndex.INDEX_UNBOUNDED) {
    throw new DownloadException("Unbounded index for representation: " + key);
}
```

But I'm not sure that it is a legit way to fix it and which problems my fix may bring.

### Reproduction steps
Try to download video with the next code:
```
File downloadFolder = new File("/storage/emulated/0/downloaded_movies/");
SimpleCache cache = new SimpleCache(downloadFolder, new NoOpCacheEvictor());

DefaultHttpDataSourceFactory factory = new DefaultHttpDataSourceFactory(userAgent);
DownloaderConstructorHelper helper = new DownloaderConstructorHelper(cache, factory);
DashDownloader dashDownloader = new DashDownloader(mpdUri, helper);
dashDownloader.download(new Downloader.ProgressListener() {
    @Override
    public void onDownloadProgress(Downloader downloader, float downloadPercentage, long downloadedBytes) {
        ...
    }
});
```

### Link to test content
https://media.axprod.net/TestVectors/v7-Clear/Manifest_1080p.mpd

### Version of ExoPlayer being used
2.6.1

### Device(s) and version(s) of Android being used
Not a device specific. I used Nexus 9 Android 7.1.1

### A full bug report captured from the device
Bug report sent by email.

**Comment by ojw28** [comment]
@erdemguven - Can we fix this? The fix suggested above looks correct to me?

**Comment by ojw28** [comment]
Fix will be in 2.7.2.
